packages/blade/src/components/Link/BaseLink/BaseLink.tsx

```diff
@@ -0,0 +1,271 @@
+import type { ReactElement, SyntheticEvent } from 'react';
+import { useState } from 'react';
+import type { CSSObject } from 'styled-components';
+import type { DurationString, EasingString } from '../../../tokens/global/motion';
+import type { LinkActionStates } from '../../../tokens/theme/theme';
+import { makeAccessible } from '../../../utils';
+import getIn from '../../../utils/getIn';
+import type { DotNotationSpacingStringToken } from '../../../_helpers/types';
+import type { Theme } from '../../BladeProvider';
+import { useTheme } from '../../BladeProvider';
+import Box from '../../Box';
+import type { IconComponent, IconProps } from '../../Icons';
+import type { BaseTextProps } from '../../Typography/BaseText';
+import BaseText from '../../Typography/BaseText';
+import StyledBaseLink from './StyledBaseLink';
+import useInteraction from './useInteraction';
+
+type BaseLinkCommonProps = {
+  intent?: 'positive' | 'negative' | 'notice' | 'information' | 'neutral';
+  contrast?: 'low' | 'high';
+  icon?: IconComponent;
+  iconPosition?: 'left' | 'right';
+  onClick?: (event: SyntheticEvent) => void;
+  accessibilityLabel?: string;
+};
+
+/*
+  Mandatory children prop when icon is not provided
+*/
+type BaseLinkWithoutIconProps = BaseLinkCommonProps & {
+  icon?: undefined;
+  children: string;
+};
+
+/*
+  Optional children prop when icon is provided
+*/
+type BaseLinkWithIconProps = BaseLinkCommonProps & {
+  icon: IconComponent;
+  children?: string;
+};
+
+/*
+  BaseLink Props with or without an icon
+*/
+type BaseLinkPropsWithOrWithoutIcon = BaseLinkWithIconProps | BaseLinkWithoutIconProps;
+
+/*
+  BaseLink Props when variant is anchor
+*/
+type BaseLinkAnchorVariantProps = BaseLinkPropsWithOrWithoutIcon & {
+  variant?: 'anchor';
+  href?: string;
+  target?: string;
+  rel?: string;
+  isDisabled?: undefined;
+};
+
+/*
+  BaseLink Props when variant is button
+*/
+type BaseLinkButtonVariantProps = BaseLinkPropsWithOrWithoutIcon & {
+  variant?: 'button';
+  isDisabled?: boolean;
+  href?: undefined;
+  target?: undefined;
+  rel?: undefined;
+};
+
+/*
+  BaseLink Props when variant is anchor or button
+*/
+export type BaseLinkProps = BaseLinkAnchorVariantProps | BaseLinkButtonVariantProps;
+
+type BaseLinkStyleProps = {
+  as: 'a' | 'button';
+  textDecorationLine: 'underline' | 'none';
+  iconColor: IconProps['color'];
+  iconPadding: DotNotationSpacingStringToken;
+  textColor: BaseTextProps['color'];
+  focusRingColor: string;
+  motionDuration: DurationString;
+  motionEasing: EasingString;
+  cursor: CSSObject['cursor'];
+  disabled: boolean;
+  role: 'button' | 'link';
+};
+
+const getColorToken = ({
+  variant,
+  intent,
+  contrast,
+  element,
+  currentInteraction,
```

Comment:
nit: why not call it `state`? after all it is hover state, active state, etc.

Reply:
Haha yes. That was my initial thought too. 

This is how it the hook looks right now

```jsx
  const { currentInteraction, setCurrentInteraction, ...syntheticEvents } = useInteraction();
```

If I want to change it to `state` it would cause a conflict with `useState`

```jsx
  const { state, setState, ...syntheticEvents } = useState();
```